src/date/index.ts

```diff
@@ -27,3 +31,378 @@ export function getMonthAbbrByIndex(monthNumber: number): string {
+ * isValidDate('2020-2-29');  // Output is true as 2020 was a leap year
+ * ```
+ */
+export function isValidDate(dateStr: string, delimiter: string = '-') {
+  try {
+    if (dateStr) {
+      const bits = dateStr.split(delimiter);
+      const date = new Date(bits[ 0 ], bits[ 1 ] - 1, bits[ 2 ]);
+
+      return date && (date.getMonth() + 1) === Number(bits[ 1 ]);
+
+    } else {
+      return false;
+    }
+
+  } catch (err) {
+    console.error('Error in validating date', err);
+    return false;
+  }
+}
+
+
+/**
+ * This function returns an array of objects which specify financial years from last supported year to offset
+ * year from current financial year.
+ *
+ * @remarks
+ * This function returns a list of UI data to be returned from a start year to end year (financial year data).
+ * lastSupportedyear is the start year, and the end year is current year or if any offset is there, then end year is current year
+ * minus offset. This function helps in showing a dropdown of all available financial years to choose from.
+ * Mainly used in Reports section of the website.
+ *
+ * @param {number} lastSupportedYear - last year in number from which array of values needs to be returned
+ * @param {number} offsetFromCurrentFinancialYear - offset from current year of the year upto which the values have to be returned
+ *
+ * @example
+ * ```
+ * getReportDateInput(2020, 1);  // Output will be [{startDate: 'Apr 2020', endDate: 'Mar 2021'}]
+ * getReportDateInput(2020);  // Output will be [{startDate: 'Apr 2021', endDate: 'Mar 2022'}, {startDate: 'Apr 2020', endDate: 'Mar 2021'}]
+ * ```
+ */
+export function getReportDateInput(lastSupportedYear: number, offsetFromCurrentFinancialYear: number = 0) {
+  try {
+    const dateInput = [];
+    const currentYear = new Date().getFullYear();
+    const currentMonth = new Date().getMonth();
+    const currentFinancialYear = currentMonth < 4
+      ? currentYear - 1 - offsetFromCurrentFinancialYear
+      : currentYear - offsetFromCurrentFinancialYear;
+
+    for (let financialYear = currentFinancialYear; financialYear >= lastSupportedYear; financialYear--) {
+      const dateEntry = {
+        startDate: `Apr ${financialYear}`,
+        endDate: `Mar ${financialYear + 1}`
+      };
+
+      dateInput.push(dateEntry);
+    }
+
+    return dateInput;
+
+  } catch (error) {
+    console.error('Report-date generation error:', error);
+    return [];
+  }
+}
+
+
+/**
+ * This method inserts "/" while entering dates in input element
+ *
+ * @param {string} inputDate - string entered in input element
+ *
+ * @example
+ * ```
+ * formatDateWithBackSlash('121');  // Output will be '12/1'
+ * formatDateWithBackSlash('12');  // Output will be '12/'
+ * formatDateWithBackSlash('1');  // Output will be '1'
+ * formatDateWithBackSlash('1213');  // Output will be '12/13'
+ * formatDateWithBackSlash('12121212');  // Output will be '12/12/1212'
+ * ```
+ */
+export function formatDateWithBackSlash(inputDate: string) {
```

Comment:
For use cases like these.. Using text input type as date is much easier. It handles it perfectly. 

Also is this happening while the user is inputting?

Reply:
we are using string methods here like replace, slice etc thats why using string type.

Yes this function is called while user is inputting. Adding a remark in the comments for the same.